### PR TITLE
install roles from galaxy

### PR DESCRIPTION
**Pull request by pampatzoglou**

### CODE REVIEW

**Comment by fgimenez** [comment]
An actual e2e test would be very helpful.

**Comment by fgimenez** [comment]
We also need to be able to define in the config which version of the collection we want to use (even also which collection); this way we can publish new versions without breaking people using the tool.

File: src/lib/clients/ansible.js [hunk 1]
```diff
@@ -20,6 +20,14 @@ class Ansible {
   }
 
   async sync() {
+    const util = require('util');
+    const exec = util.promisify(require('child_process').exec);
+
+    async function ls() {
```
Comment (fgimenez) [inline src/lib/clients/ansible.js:26]: No please, use `this._cmd` for calling commands, the same as the rest of the module does.

File: src/lib/clients/ansible.js [hunk 2]
```diff
@@ -20,6 +20,14 @@ class Ansible {
...
+      const { stdout, stderr } = await exec('ansible-galaxy collection install --force -p ../../../ansible w3f.polkadot_validator:0.0.3');
```
Comment (fgimenez) [inline src/lib/clients/ansible.js:27]: The version of the collection needs to be defined as a configuration value, having it hardcoded makes things much harder to manage.

File: src/lib/clients/ansible.js [hunk 3]
```diff
@@ -20,9 +20,12 @@ class Ansible {
   }
 
   async sync() {
+    const util = require('util');
+    const exec = util.promisify(require('child_process').exec);
```
Comment (fgimenez) [inline src/lib/clients/ansible.js:24]:
```suggestion
```

**Comment by pampatzoglou** [comment]
closes https://github.com/w3f/infrastructure/issues/38

File: .eslintrc.json [hunk 4]
```diff
@@ -15,6 +15,7 @@
   },
   "rules": {
     "no-console": 0,
-    "no-useless-escape": 0
+    "no-useless-escape": 0,
+    "no-async-promise-executor": 0
```
Comment (fgimenez) [inline .eslintrc.json:19]:
```suggestion
    "no-useless-escape": 0
```

File: src/lib/clients/ansible.js (hunk 2 above)
Comment (pampatzoglou) [reply src/lib/clients/ansible.js]: executed here:
https://github.com/w3f/polkadot-secure-validator/blob/cc0d47b43e8b4aa063db4870286518aaf4594baa/src/lib/clients/ansible.js#L24-L26

value here:
https://github.com/w3f/polkadot-secure-validator/blob/cc0d47b43e8b4aa063db4870286518aaf4594baa/config/main.sample.json#L15
wdyt?

**Comment by pampatzoglou** [comment]
https://app.circleci.com/pipelines/github/w3f/polkadot-secure-validator/257/workflows/722e8bf2-ea43-4aca-84e5-618572f74b77/jobs/1682

**Comment by fgimenez** [comment]
Outdated, closing

**Pull request closed without merging**